flatflow/data/dataset_test.cc

```diff
@@ -179,4 +203,165 @@ TEST(DatasetTest, IntraBatchShuffling) {
   }
 }
 
+TEST(DatasetTest, IndexRetriever) {
+  // Note:
```

Comment:
Please use upper case for comments to be rendered - i.e., ``NOTE:`` instead of ``Note:``.